classes/battle/BattleManager.php

```diff
@@ -723,7 +725,31 @@ protected function setupFighterAttack(Fighter $fighter, LegacyFighterAction $act
                 break;
         }
 
-        $attack->raw_damage = $fighter->calcDamage(attack: $attack->jutsu, disable_randomness: $disable_randomness);
+        // Setup clash effects
+        foreach ($attack->jutsu->effects as $effect) {
+            switch ($effect->effect) {
+                case 'piercing':
+                    $attack->piercing_percent += $effect->effect_amount / 100;
+                    break;
+                case 'substitution':
+                    $attack->substitution_percent += $effect->effect_amount / 100;
+                    break;
+                case 'counter':
+                    $attack->counter_percent += $effect->effect_amount / 100;
+                    break;
+                case 'recoil':
+                    $attack->recoil_percent += $effect->effect_amount / 100;
+                    break;
+                case 'immolate':
+                    $attack->immolate_percent += $effect->effect_amount / 100;
+                    $attack->immolate_raw_damage += $this->effects->processImmolate($attack, $target);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        $attack->raw_damage = $fighter->calcDamage(attack: $attack->jutsu, disable_randomness: $disable_randomness, immolate_raw_damage: $attack->immolate_raw_damage * $attack->immolate_percent);
```

Comment:
```suggestion
        $attack->raw_damage = $fighter->calcDamage(attack: $attack->jutsu, disable_randomness: $disable_randomness, immolate_raw_damage: $attack->immolate_raw_damage * $attack->immolate_percent);
```
Wait, won't this cause immolate to double-dip? It gets added to the attack's raw damage here, and then also triggers separately in `applyAttack`

I think the `applyAttack` is the one we should keep and this is the one we should drop, unlike recoil immolate is not making your attack itself more powerful, it triggers a secondary burst of damage from bleeding/fire/etc that is already on the target